bootstrap/templates/ansible/playbooks/cluster-rollout-update.yaml.j2

```
@@ -25,46 +25,48 @@
         # If spec.unschedulable is defined then the node is cordoned
         - not (kubectl_get_node['stdout'] | from_json).spec.unschedulable is defined
       block:
-        - name: Cordon node
-          ansible.builtin.command: "kubectl cordon {% raw %}{{ inventory_hostname }}{% endraw %}"
+        - name: "Cordon {% raw %}{{ inventory_hostname }}{% endraw %}"
+          kubernetes.core.k8s_drain:
+            name: "{% raw %}{{ inventory_hostname }}{% endraw %}"
+            kubeconfig: /etc/rancher/k3s/k3s.yaml
+            state: cordon
           delegate_to: "{% raw %}{{ groups['master'][0] }}{% endraw %}"
-          changed_when: false
 
-        - name: Wait for node to cordon
-          ansible.builtin.command: "kubectl get node {% raw %}{{ inventory_hostname }}{% endraw %} -o json"
-          register: wait_for_cordon
-          retries: 10
-          delay: 10
+        - name: "Drain {% raw %}{{ inventory_hostname }}{% endraw %}"
+          kubernetes.core.k8s_drain:
+            name: "{% raw %}{{ inventory_hostname }}{% endraw %}"
+            kubeconfig: /etc/rancher/k3s/k3s.yaml
+            state: drain
+            delete_options:
+              delete_emptydir_data: true
+              ignore_daemonsets: true
+              terminate_grace_period: 300
+              wait_timeout: 600
```

```suggestion
              terminate_grace_period: 600
              wait_timeout: 900
              force: true
```

I think this should be good, some pods (loki) can take up-to 10 minutes to be killed.